### PR TITLE
Changes for integration of SGX attestation

### PR DESCRIPTION
**Pull request by dgoltzsche**
Main changes:
- additional fields in proto message
- support for encrypted wasm files
- base64-encoded encrypted results

### CODE REVIEW

**Comment by csegarragonz** [comment]
Hey @dgoltzsche,

There was a bug with our compose files after GH's update to `1.28.4` that made our scripts fail. Would you mind rebasing to the latest master as soon as #58 is merged?

File: src/proto/faabric.proto [hunk 1]
```diff
@@ -108,6 +108,13 @@ message Message {
     string cmdline = 39;
 
     bool isSgx = 40;
+
+    // fields for SGX attesation
+    string sid = 41;
+    string nonce = 42;
+    string tag = 43;
+    bytes policy = 44;
+    bytes result = 45;
```
Comment (Shillaker) [inline src/proto/faabric.proto:117]: Could we prepend `sgx_` to all of these fields?

File: src/util/func.cpp [hunk 2]
```diff
@@ -202,6 +203,13 @@ std::string getFunctionFile(const faabric::Message& msg)
     return path.string();
 }
 
+std::string getEncryptedFunctionFile(const faabric::Message& msg)
+{
+    auto path = getFunctionDir(msg);
+    path.append(encryptedFuncFile);
+
+    return path.string();
+}
```
Comment (Shillaker) [inline src/util/func.cpp:212]: Think there's a missing newline here between the functions, surprised formatting didn't catch

File: src/util/json.cpp [hunk 3]
```diff
@@ -274,6 +293,8 @@ faabric::Message jsonToMessage(const std::string& jsonIn)
     msg.set_cmdline(getStringFromJson(d, "cmdline", ""));
 
     msg.set_issgx(getBoolFromJson(d, "sgx", false));
+    msg.set_sid(getStringFromJson(d, "sid", ""));
+    msg.set_tag(getStringFromJson(d, "tag", ""));
```
Comment (Shillaker) [inline src/util/json.cpp:297]: Can you add these fields to the test for the message json round trip?

File: src/util/json.cpp [hunk 4]
```diff
@@ -160,6 +162,23 @@ std::string messageToJson(const faabric::Message& msg)
...
+    d.AddMember("result", Value(result_.c_str(), result_.size(), a).Move(), a);
+    d.AddMember(
+      "output_data",
+      Value(msg.outputdata().c_str(), msg.outputdata().size(), a).Move(),
+      a);
+    StringBuffer sb;
+    Writer<StringBuffer> writer(sb);
+    d.Accept(writer);
+    return sb.GetString();
+}
```
Comment (Shillaker) [inline src/util/json.cpp:180]: Could you add a small test for this? Can just go alongside the usual `util` tests

**Review by Shillaker — CHANGES_REQUESTED** [review]

**Review by Shillaker — APPROVED** [review]

**Pull request merged**